src/aiida/cmdline/commands/cmd_group.py

```diff
@@ -145,41 +145,164 @@ def group_move_nodes(source_group, target_group, force, nodes, all_entries):
+        if endswith:
+            filters['or'].append({'label': {'like': f'%{escape_for_sql_like(endswith)}'}})
+        if contains:
+            filters['or'].append({'label': {'like': f'%{escape_for_sql_like(contains)}%'}})
+
+        builder.append(orm.Group, filters=filters, tag='group', project='*')
+
+        # Query groups that belong to specific user
+        if user:
+            user_email = user.email
+        else:
+            # By default: only groups of this user
+            user_email = orm.User.collection.get_default().email
+
+        # Query groups that belong to all users
+        if not all_users:
+            builder.append(orm.User, filters={'email': {'==': user_email}}, with_group='group')
+
+        # Query groups that contain a particular node
+        if node:
+            builder.append(orm.Node, filters={'id': {'==': node.pk}}, with_group='group')
+
+        groups = builder.all()
+
+        groups = reduce(concat, groups)
+
+        if not groups:
+            echo.echo_report('No groups found matching the specified criteria.')
             return
 
-    if not dry_run:
+    if groups:
+        projection_lambdas = {
+            'pk': lambda group: str(group.pk),
+            'label': lambda group: group.label,
+            'type_string': lambda group: group.type_string,
+            'count': lambda group: group.count(),
+            'user': lambda group: group.user.email.strip(),
+            'description': lambda group: group.description,
+        }
+
+        table = []
+        projection_header = ['PK', 'Label', 'Type string', 'User']
+        projection_fields = ['pk', 'label', 'type_string', 'user']
+        for group in groups:
+            table.append([projection_lambdas[field](group) for field in projection_fields])
+
+        if not (force or dry_run):
+            echo.echo_report('The following groups will be deleted:')
+            echo.echo(tabulate(table, headers=projection_header))
+            click.confirm('Are you sure you want to continue?', abort=True)
+        elif dry_run:
+            echo.echo_report('Would have deleted:')
+            echo.echo(tabulate(table, headers=projection_header))
+
+    for group in groups:
         group_str = str(group)
-        orm.Group.collection.delete(group.pk)
-        echo.echo_success(f'{group_str} deleted.')
+
+        if delete_nodes:
+
+            def _dry_run_callback(pks):
+                if not pks or force:
+                    return False
+                echo.echo_warning(
+                    f'YOU ARE ABOUT TO DELETE {len(pks)} NODES ASSOCIATED WITH {group_str}! THIS CANNOT BE UNDONE!'
+                )
+                return not click.confirm('Do you want to continue?', abort=True)
+
+            _, nodes_deleted = delete_group_nodes([group.pk], dry_run=dry_run or _dry_run_callback, **traversal_rules)
+            if not nodes_deleted:
+                # don't delete the group if the nodes were not deleted
+                return
```

Comment:
What if the group was empty? Will `nodes_deleted` be `False` then? If this means that empty groups are not deleted when the user specifies `--delete-nodes` that would be incorrect. This would definitely be a case to add as a unit test.

Reply:
Good point
I added, unit test. 
Apparently, `delete_group_nodes`  would return `True` in that case. So it's all good.